src/state/mod.rs

```diff
@@ -115,25 +123,82 @@ impl PersistedState {
         match action {
-            Actions::AddRemoveConversations(new_chats) => {
-                self.favorites = self
-                    .favorites
-                    .iter()
-                    .filter(|id| new_chats.contains_key(id))
-                    .cloned()
-                    .collect();
-                self.all_chats = new_chats;
-                self.total_unreads = total_notifications(&self);
+            Actions::AddConversation(conversation) => {
+                self.all_chats
+                    .insert(conversation.id(), conversation.clone());
+                self.active_chats.insert(
+                    conversation.id(),
+                    ConversationInfo {
+                        conversation,
+                        creation_time: DateTime::from(Local::now()),
+                        ..Default::default()
+                    },
+                );
+            }
+            Actions::HideChat(conversation_id) => {
+                self.active_chats.remove(&conversation_id);
+                if self.selected_chat == Some(conversation_id) {
+                    self.selected_chat = None;
+                }
+                // todo: should the hidden chat be removed from favorites too?
+                // let favorites = self
+                //     .favorites
+                //     .iter()
+                //     .filter(|id| conversation_id == **id)
+                //     .cloned()
+                //     .collect();
+                // self.favorites = favorites;
             }
-            Actions::ClearChat => {
-                self.current_chat = None;
+            Actions::ShowChat(uuid) => {
+                // look up uuid in all_chats
+                match self.all_chats.get(&uuid) {
+                    // add to active_chats
+                    Some(conv) => {
+                        self.active_chats.insert(
+                            uuid,
+                            ConversationInfo {
+                                conversation: conv.clone(),
+                                ..Default::default()
+                            },
+                        );
+                        // set selected_chat
+                        self.selected_chat = Some(uuid);
+                    }
+                    None => {
+                        log::error!("ShowChat called for nonexistent chat. uuid: {}", uuid);
+                    }
+                }
             }
-            Actions::ChatWith(info) => {
-                self.current_chat = Some(info.conversation.id());
+            //Actions::DeselectChat => {
+            //    self.selected_chat = None;
+            //}
+            Actions::ChatWith { mut rg, friend } => {
+                // hopefully there isn't a race condition between this part of code and the part that receives events for new conversations. want this to be done before that
+                // event is received.
+                let conversation =
```

Comment:
I dont think this should be here. If it must be here if there is an error besides `Error::ConversationExist` then it shouldnt attempt to use `Conversation::default` as if there is an error creating a conversation for whatever reason then the conversation would be invalid.

Reply:
thanks for catching this. I addressed the bug and moved the creation of the `Conversation` back into the `Friend` element.